app/graphql/mutations/user_languages/create_user_language.rb

```diff
@@ -5,8 +5,21 @@ class CreateUserLanguage < ::Mutations::BaseMutation
       type Types::UserLanguageType
 
       def resolve(params:)
+        user_lang_params = Hash params
+
+        return handle_duplicate_user_lang if user_language_exists?(user_lang_params)
+
         UserLanguage.create(params.to_hash)
       end
+
+      def user_language_exists?(params)
+        !UserLanguage.where(user: params[:user_id], language: params[:language_id],
+                            fluency_level: params[:fluency_level]).empty?
```

Comment:
This logic is kinda confusing. Would it be better to use the `.exists?` method or is that only for id? Also, would this code function the same if we removed the `!` and `.empty?`

Reply:
@lkriffell the return value for the where query is an empty array so getting rid of the `!` and `.empty?` wouldn't return a boolean. I haven't seen `exists?` before though! Thanks for showing me that. I will push up now.

Reply:
I saw that create availability could also use `.exists?` so I pushed up that refactor as well!